streams/src/main/java/org/apache/kafka/streams/kstream/JoinWindows.java

```diff
@@ -168,6 +172,50 @@ public JoinWindows grace(final Duration afterWindowEnd) throws IllegalArgumentEx
         return new JoinWindows(beforeMs, afterMs, afterWindowEndMs);
     }
 
+
+    /**
+     * Specifies that records of the same key are joinable if their timestamps are within {@code timeDifference},
+     * i.e., the timestamp of a record from the secondary stream is max {@code timeDifference} earlier or later than
+     * the timestamp of the record from the primary stream.
+     *
+     * @param timeDifference join window interval
+     * @throws IllegalArgumentException if {@code timeDifference} is negative or can't be represented as {@code long milliseconds}
+     * @since 3.0
+     */
+    public static JoinWindows ofTimeDifferenceWithNoGrace(final Duration timeDifference) {
+        return ofTimeDifferenceAndGrace(timeDifference, ofMillis(DEFAULT_GRACE_PERIOD_MS));
```

Comment:
I think it would be more clear to specify this as follows. The whole point of this KIP is to get rid of the concept of a "default grace period" 
```suggestion
        return ofTimeDifferenceAndGrace(timeDifference, ofMillis(0));
```

Reply:
@ableegoldman this makes sense. I dont really like Magic numbers so I can create something like NO_GRACE_PERIOD (with a value of 0L) instead. But I appreciate this feedback.